hax/hax/bytecount.py

```diff
@@ -113,8 +113,9 @@ def _get_parity_buffers(self, bc: int, state: PverInfo) -> int:
         Calculate the parity buffer count based on pool configuration.
         """
         tot_units = state.data_units + state.parity_units
-        bytes_per_unit = ceil(bc / tot_units)
-        return bytes_per_unit * state.parity_units
+        percent = state.parity_units / tot_units
+        LOG.debug('Parity buffer percentage %s', percent)
+        return ceil(bc * percent)
```

Comment:
Is ceiling is a must here ?

Reply:
while calculating parity percentage, we get results in floats.
```
$ hctl status
Bytecount:
critical : 0
damaged : 0
degraded : 0
healthy : 34952533.333333336
```

Confirmed with @yeshpal-jain-seagate  and @AbhishekSahaSeagate for using `ceil` here. Let me know, if any change is required.

Reply:
okay, fine.

Reply:
ceiling is required as the result may be a fraction.

Reply:
@madhavemuri, why do we need percentage here?
So the earlier algorithm would work the same way, consider the following example with partially populated parity group, with unit size of 1M, N = 4, K = 2 and data written = 9M
we have 9M of data and 6M of parity = 15M of total bytecount, now we do,

scale down the entire bytecount and virtaully concentrate it in a single parity group, ceil(15/6) = 3M (virtual temporary unit size)
Now we do, K * 3M = 6M
actual bytecount = 15M - 6M = 9M
So the above calculate will work for both partially populated and fully populated parity groups, how would using percentage help to do it any better? cc @Shreya-18

Reply:
how do we know what is the unit size for that stripe, we do not maintain any unit size parameter for a particular size.

Reply:
@yeshpal-jain-seagate, why do we need to know the unit size? Here we are reducing multiple parity groups to a single group and extending the unit size to a new virtual unit size as we just need to separate out the parity from actual data.